sdk/core/azure-core/src/base64.cpp

```diff
@@ -490,7 +490,7 @@ namespace Azure { namespace Core {
 
   std::string Convert::Base64Encode(const std::vector<uint8_t>& data)
   {
-    return ::Base64Encode(data);
+    return ::Base64Encode(data.data(), data.size());
```

Comment:
data.size() or length ? you replaced it before

Reply:
The idea here is that the underlying `::Base64Encode` encodes a pointer+length value.

`Convert::Base64Encode` has specializations for the various things we want to encode, all of which call into `::Base64Encode`.

This allows us to support Base64 encoding data without additional trips through the memory allocator.